packages/fileeditor-extension/src/commands.ts

```diff
@@ -0,0 +1,888 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { JupyterFrontEnd } from '@jupyterlab/application';
+
+import { ICommandPalette, WidgetTracker } from '@jupyterlab/apputils';
+
+import { CodeEditor } from '@jupyterlab/codeeditor';
+
+import { IConsoleTracker } from '@jupyterlab/console';
+
+import {
+  ISettingRegistry,
+  MarkdownCodeBlocks,
+  PathExt
+} from '@jupyterlab/coreutils';
+
+import { IDocumentWidget } from '@jupyterlab/docregistry';
+
+import { IFileBrowserFactory } from '@jupyterlab/filebrowser';
+
+import { FileEditor } from '@jupyterlab/fileeditor';
+
+import { ILauncher } from '@jupyterlab/launcher';
+
+import {
+  IEditMenu,
+  IFileMenu,
+  IMainMenu,
+  IRunMenu,
+  IViewMenu
+} from '@jupyterlab/mainmenu';
+
+import { CommandRegistry } from '@phosphor/commands';
+
+import { JSONObject, ReadonlyJSONObject } from '@phosphor/coreutils';
+
+import { Menu } from '@phosphor/widgets';
+
+/**
+ * The command IDs used by the fileeditor plugin.
+ */
+export namespace CommandIDs {
+  export const createNew = 'fileeditor:create-new';
+
+  export const createNewMarkdown = 'fileeditor:create-new-markdown-file';
+
+  export const changeFontSize = 'fileeditor:change-font-size';
+
+  export const lineNumbers = 'fileeditor:toggle-line-numbers';
+
+  export const lineWrap = 'fileeditor:toggle-line-wrap';
+
+  export const changeTabs = 'fileeditor:change-tabs';
+
+  export const matchBrackets = 'fileeditor:toggle-match-brackets';
+
+  export const autoClosingBrackets = 'fileeditor:toggle-autoclosing-brackets';
+
+  export const createConsole = 'fileeditor:create-console';
+
+  export const runCode = 'fileeditor:run-code';
+
+  export const runAllCode = 'fileeditor:run-all';
+
+  export const markdownPreview = 'fileeditor:markdown-preview';
+}
+
+/**
+ * The class name for the text editor icon from the default theme.
+ */
+export const EDITOR_ICON_CLASS = 'jp-MaterialIcon jp-TextEditorIcon';
+
+/**
+ * The class name for the text editor icon from the default theme.
+ */
+export const MARKDOWN_ICON_CLASS = 'jp-MarkdownIcon';
+
+/**
+ * The name of the factory that creates editor widgets.
+ */
+export const FACTORY = 'Editor';
+
+/**
+ * A utility class for adding commands and menu items,
+ * for use by the File Editor extension or other Editor extensions.
+ */
+export default class Commands {
```

Comment:
Can you make this a namespace instead of a class? It seems that since everything is mostly static functions, it makes more sense to just make it a namespace. Then the private things below just don't get exported, and the normal static functions just become exported functions.

Reply:
I've updated it to a namespace, it required some tinkering with the private content, but it made all the exported content much cleaner (no more `this.`)